src/Parser.ts

```diff
@@ -66,26 +97,29 @@ export class Parser {
     } else if (this.match(TokenType.NUMBER)) {
       return this.num()
     }
-    throw(new ParserError("unexpected primary parse token ${peek().type}"))
+    throw new ParserError(`unexpected primary parse token ${this.peek().ttype}`)
   }
 
   num(): Num {
     return new Num(this.previous().literal)
   }
 
   group(): Expression {
-      const expr = this.expression()
-      this.consume(TokenType.RIGHT_PAREN)
-      return expr
+    const expr = this.expression()
+    this.consume(TokenType.RIGHT_PAREN)
+    return new Group(expr)
   }
 
   // END GRAMMAR FUNCTIONS
 
   match(ttype: TokenType): Boolean {
-    if (this.check(ttype)) {
-      return true
+    while (!this.isAtEnd()) {
```

Comment:
why the while statement?

Reply:
I think I wrote this back when I had a less of an understanding of the difference between if and whiles. I can see now that there is no reason for it to do a looping check to see if it's at the end within this code.